### PR TITLE
Add groupAC, noAssertion, SOAIdentifier, and noRevAvail exts

### PR DESCRIPTION
**Pull request by JonathanWilbur**
Signed-off-by: Jonathan M. Wilbur <jonathan@wilbur.space>

This PR adds the `groupAC`, `noAssertion`, `sOAIdentifier`, and `noRevAvail` certificate extensions from the International Telecommunication Union's Recommendation X.509. These extensions are used for attribute certificates, whose [implementation in OpenSSL is in progress](https://github.com/openssl/openssl/pull/15857).

I added all of these extensions in a single PR because they all have the syntax `NULL`, so it was more-or-less copy-and-paste to make them all work. I also did this to accelerate their incorporation into OpenSSL.

This PR also relates to [this pull request](https://github.com/openssl/openssl/pull/16462) I submitted to include the `singleUse` attribute certificate extension. That PR already has one approval.

##### Checklist

- [x] tests are added or updated

### CODE REVIEW

**Comment by t8m** [comment]
We are sorry for neglecting this. Can you please rebase the PR to the fresh master branch to resolve conflicts?

File: include/openssl/obj_mac.h [hunk 1]
```diff
@@ -2634,7 +2634,7 @@
 #define OBJ_crl_reason          OBJ_id_ce,21L
 
 #define SN_invalidity_date              "invalidityDate"
-#define LN_invalidity_date              "Invalidity Date"
+#define LN_invalidity_date              "X509v3 Invalidity Date"
```
Comment (t8m) [inline include/openssl/obj_mac.h:2637]: This would be an API break. Please revert this change.

File: crypto/x509/v3_group_ac.c [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+#include "internal/cryptlib.h"
+#include <openssl/asn1.h>
+#include <openssl/asn1t.h>
+#include <openssl/x509v3.h>
+#include "ext_dat.h"
+
+static int i2r_GROUP_AC(X509V3_EXT_METHOD *method,
+                        void *su, BIO *out,
+                        int indent);
+
```
Comment (t8m) [inline crypto/x509/v3_group_ac.c:20]: Why this duplication?

Comment (JonathanWilbur) [reply crypto/x509/v3_group_ac.c]: That was a mistake on my part. I think because I copied a lot of code between files in this commit.

**Comment by JonathanWilbur** [comment]
@t8m Just checking: you aren't waiting on me for anything, are you?

File: crypto/x509/v3_no_ass.c [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+#include <stdio.h>
+#include "internal/cryptlib.h"
+#include <openssl/asn1.h>
+#include <openssl/asn1t.h>
+#include <openssl/x509v3.h>
+#include "ext_dat.h"
+
+static int i2r_NO_ASSERTION(X509V3_EXT_METHOD *method,
+                            void *su, BIO *out,
+                            int indent);
```
Comment (t8m) [inline crypto/x509/v3_no_ass.c:19]: Please remove this extra declaration.

File: crypto/x509/v3_no_rev_avail.c [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+#include "internal/cryptlib.h"
+#include <openssl/asn1.h>
+#include <openssl/asn1t.h>
+#include <openssl/x509v3.h>
+#include "ext_dat.h"
+
+static int i2r_NO_REV_AVAIL(X509V3_EXT_METHOD *method,
+                            void *su, BIO *out,
+                            int indent);
+
```
Comment (t8m) [inline crypto/x509/v3_no_rev_avail.c:20]: Please remove this as well.

File: crypto/x509/v3_soa_id.c [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+#include <stdio.h>
+#include "internal/cryptlib.h"
+#include <openssl/asn1.h>
+#include <openssl/asn1t.h>
+#include <openssl/x509v3.h>
+#include "ext_dat.h"
+
+static int i2r_SOA_IDENTIFIER(X509V3_EXT_METHOD *method,
+                              void *su, BIO *out,
+                              int indent);
```
Comment (t8m) [inline crypto/x509/v3_soa_id.c:19]: And this as well.

**Review by t8m — CHANGES_REQUESTED** [review]
The commit message talks about the added extensions but there are also some OBJECTs added. Is that intentional? In that case the commit message should be expanded and/or the objects added should be a separate commit.

**Comment by JonathanWilbur** [comment]
@t8m I have addressed the outstanding code review issues and re-worded the first commit to mention the added attribute certificate extension object identifiers.
